### ISSUE TITLE
Null coalesce invalid detect / fix with reference

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#1105]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug / Feature request                          |
| Plugin         | Php Inspections (EA Extended), 3.0.8.1 |
| Language level | PHP 7.0      |

This

```
    $_dept      = &Request::$r->_dept;

    if (isset($original_dept)) {
        $_dept = $original_dept;
    }
```

Was fixed to

```
    $_dept = $original_dept ?? Request::$r->_dept;
```

But that fix is invalid since $_dept is a reference.

A correct fix could be

```
Request::$r->_dept = $original_dept ?? Request::$r->_dept;
```

although I'm not sure I'd want that.  Can you think of anything better?  Or should this just be ignored?

**Comment by iakoder** [comment]
I was just looking at this code again and noticed that only one of the two isset is being caught in the full version of the code I originally provided.  I can see that it's not quite the pattern you're looking for.

````
    $_location  = &Request::$r->_location;
    $_dept      = &Request::$r->_dept;

    if (isset($original_dept)) {
        $_dept = $original_dept;
    }
    if (isset($original_location)) {
        $_location = $original_location;
    }
````

**Comment by kalessil** [comment]
Indeed, that's not a target pattern. The inspection was not intended to work with references - semantics guesses are hardly possible.

**Comment by kalessil** [comment]
Fixed!
